### DISCUSSION TITLE
Direct Streamline Normalization ，The result is wrong

### DISCUSSION THREAD

**Discussion by wangyibin0011** [dipy/dipy#2167] [Tractography / Tractometry]
Hello, when I tried the example of Dipy
——References
Direct Streamline Normalization, the result was very strange when visualizing. Is this a problem of data or other problems?
![1](https://user-images.githubusercontent.com/53501301/81631566-b7bf4100-943a-11ea-9bb7-d3b42f1d51f0.png)
[lr-superiorfrontal.zip](https://github.com/dipy/dipy/files/4613064/lr-superiorfrontal.zip)

**Comment by arokem** [comment]
Is this related somehow to your issue in #2165? Is this the same data?

**Comment by Garyfallidis** [comment]
It would help if you could give us a bit more information so we can understand what you are trying to do. It would help even more if you could provide some sample data to showcase the problem.

**Comment by wangyibin0011** [comment]
> 这与您在[＃2165中](https://github.com/dipy/dipy/issues/2165)的问题有某种联系吗？这是相同的数据吗？
Lr -superiorfrontal. TRK is what I generated when I ran this example:
https://dipy.org/documentation/1.1.1./examples_built/streamline_tools/#streamline-tools
Then I ran this example [https://www.dipy.org/documentation/1.1.1./examples_built/streamline_registration/#example-streamline-registration] to look at the registration of T2 and b0, which was fine, but when I imported lr-superiorfrontal. TRK, this example didn't seem to get the effect I wanted.

**Comment by wangyibin0011** [comment]
> > 这与您在[＃2165中](https://github.com/dipy/dipy/issues/2165)的问题有某种联系吗？这是相同的数据吗？
> > Lr -superiorfrontal。TRK是我运行此示例时生成的：
> > [https](https://dipy.org/documentation/1.1.1./examples_built/streamline_tools/#streamline-tools) : [//dipy.org/documentation/1.1.1./examples_built/streamline_tools/#streamline-tools](https://dipy.org/documentation/1.1.1./examples_built/streamline_tools/#streamline-tools) 
> > 然后运行此示例[https://www.dipy.org/文档/1.1.1./examples_built/streamline_registration/#example-streamline-registration]来查看T2和b0的注册，这很好，但是当我导入lr-superiorfrontal时。TRK，这个例子似乎并没有达到我想要的效果。

是因为我选择的trk文件不正确吗

**Comment by wangyibin0011** [comment]
> 这与您在[＃2165中](https://github.com/dipy/dipy/issues/2165)的问题有某种联系吗？这是相同的数据吗？
Hello, I am looking for a solution for the previous problem. This example seems to solve part of my problem. I obtained the data according to the above description of the example

**Comment by dPys** [comment]
@wangyibin0011  Were you ever able to solve this? Once upon a time I spent countless hours on this issue, and the Direct Streamline Normalization tutorial probably needs to be updated at some point.

For awhile I thought I had accommodated for all cases by using a multiplicative Van Der Corput sequence to learn the FOV transforms needed across different voxel resolutions. Things can still break, however, depending on how the streamlines were saved and loaded. The orientations and resolutions of template image and the B0/FA moving image, if they are different, can also impact things. The streamlines should also have been saved and loaded as .trk files, ideally in voxel-space.  

One easier way to test things can be to save the streamlines as a density map, so that you can more easily view the overlays and avoid having to re-render the streamlines in a 3D-viewer. See https://github.com/dPys/PyNets/blob/master/pynets/registration/register.py#L333-L420 (remains unfinished) as a starting point.
